lnwallet/channel.go

```diff
@@ -2430,23 +2430,27 @@ func (lc *LightningChannel) createCommitmentTx(c *commitment,
 	}
 
 	var (
-		delay                      uint32
-		delayBalance, p2wkhBalance btcutil.Amount
+		localCfg, remoteCfg         *channeldb.ChannelConfig
+		localBalance, remoteBalance btcutil.Amount
 	)
 	if c.isOurs {
-		delay = uint32(lc.localChanCfg.CsvDelay)
-		delayBalance = ourBalance.ToSatoshis()
-		p2wkhBalance = theirBalance.ToSatoshis()
+		localCfg = lc.localChanCfg
+		remoteCfg = lc.remoteChanCfg
+		localBalance = ourBalance.ToSatoshis()
+		remoteBalance = theirBalance.ToSatoshis()
 	} else {
-		delay = uint32(lc.remoteChanCfg.CsvDelay)
-		delayBalance = theirBalance.ToSatoshis()
-		p2wkhBalance = ourBalance.ToSatoshis()
+		localCfg = lc.remoteChanCfg
+		remoteCfg = lc.localChanCfg
+		localBalance = theirBalance.ToSatoshis()
+		remoteBalance = ourBalance.ToSatoshis()
 	}
```

Comment:
Would it be better to have two different calls to `CreateCommitTx`, on in each `if` block? To prevent assigning to those intermediate vars of which the name may be confusing perhaps.

Reply:
Yeah, I think that's nice. Done: https://github.com/lightningnetwork/lnd/pull/3829/commits/bfd026cf5e88b24710183d6322586767d8bae452

Reply:
I guess that is reverted now

Reply:
Ok, it is in the next commit